### PR TITLE
Add troubleshooting instructions for agents.md re: Issue #11459

### LINKED ISSUE #11459: Agent does not continue working on pool after keyboard interrupt if there are pending tasks
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

If you keyboard interrupt the prefect agent then try to start it again, any tasks that were pending will block the pool's queue until you clear them. 

### Reproduction

```python3
1. Stop an agent working on a pool with a keyboard interrupt
2. Have some pending tasks on the agent
3. Restart the agent
4. No profit....
```

### Error

```python3
The pool will not continue to be worked and the agent will sit idle
```

### Versions

```Text
Version:             2.14.11
API version:         0.8.4
Python version:      3.11.7
Git commit:          e6d7d76d
Built:               Thu, Dec 14, 2023 5:45 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         server
```

### Additional context

This is not exactly a bug, but rather something that needs to be documented with the right vocabulary for people to find it if it happens to them. Adding here for prosperity.

### PR DESCRIPTION
**Pull request by 1beb**
Added a troubleshooting note #11459

### Checklist

Closes #11459 

- [x] This pull request references any related issue by including "closes `<link to issue>`"
- [x] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

**Review by discdiver — APPROVED** [review]
Thank you!

File: docs/concepts/agents.md [hunk 1]
```diff
@@ -84,3 +84,9 @@ This example will poll every work queue that starts with "foo-".
 By default, the agent begins submission of flow runs a short time (10 seconds) before they are scheduled to run. This allows time for the infrastructure to be created, so the flow run can start on time. In some cases, infrastructure will take longer than this to actually start the flow run. In these cases, the prefetch can be increased using the `--prefetch-seconds` option or the `PREFECT_AGENT_PREFETCH_SECONDS` setting.
 
 Submission can begin an arbitrary amount of time before the flow run is scheduled to start. If this value is _larger_ than the amount of time it takes for the infrastructure to start, the flow run will _wait_ until its scheduled start time. This allows flow runs to start exactly on time.
+
+## Troubleshooting
+
+### Agent crash or keyboard interrupt
```
Comment (serinamarie) [inline docs/concepts/agents.md:90]: @discdiver is this meant to be `on` by chance?

Comment (1beb) [reply docs/concepts/agents.md:90]: It happens if you kill (like -9) the agent while tasks are pending as well. Hence the wording for "is ended abruptly".

Comment (serinamarie) [reply docs/concepts/agents.md:90]: I'm just wondering if it's supposed to be `on` instead of `or` :)

Comment (1beb) [reply docs/concepts/agents.md:90]: Definitely `or`. Feel free to wordsmith!

Comment (serinamarie) [reply docs/concepts/agents.md:90]: cool, just checking! thanks

**Pull request merged**